### ISSUE TITLE
Add 

### ISSUE THREAD

**Issue by ragulpr** [ragulpr/taildropout#14]
I forgot where the derivation of solving the integral equation of the survival function $p=\int_0^1 S(x) dx$ to find the parameters of the distribution that generates the cutoff-`K`'s.

https://github.com/ragulpr/taildropout/blob/b667b5aa92b3bbd0fbcae9a041138b9ab174cce0/taildropout.py#L8-L18

I haven't worked on this in 8 or so years, and when I was working on this the last time I spent 24h/day thinking in terms of waiting time distributions so it must have been obvious to me then. To add to this, my notes and draft paper on this is on a harddrive overseas. 

I managed to figure out a post-hoc proof of it though:

Assume we want the fraction of "ones" in the mask to be `p`, i.e `p_obs ~= p`:
```
mask = TailDropout(p)(torch.ones([n_batch,n_features]))
p_obs = mask.mean()
```
Here we want to construct `Prob(mask[:,k]==1)` to have that property. Note that we can think of the random choice of `k` as 

$\underline{\overline{K}}_b$=`mask[b,:].sum(-1)` 

Or think of the having to choose `k` from `[0,1,..,n-1]`, you can think of this as a positive random variable that is truncated to `n`. Without loss of generality, transform the interval `[0,n]` to `[0,1]` then $\underline{\overline{K}} = min(\underline{\overline{X}},1)$.

We want to find the distribution s.t $E[\underline{\overline{K}}] = p$ so we solve

$$
p=E[\underline{\overline{K}}] = \int_0^1 x f(x) dx +  \int_1^\infty 1 f(x) dx =    \int_0^1 x f(x) dx  dx + S(1)
$$

Using

$$
\dfrac{\partial}{\partial x} x F(x) = F(x) + x f(x)
$$

$$
\int_0^1 x f(x) dx = 
\int_0^1 \left[ \dfrac{\partial}{\partial x} x F(x)  -F(x) \right]dx = 
x F(x) \Big|_0^1 \ - \int_0^1 F(x) = 
F(1) - \int_0^1 F(x)
$$

We arrive at

$$
p=
\int_0^1 x f(x) dx  dx + S(1) = 
F(1) - \int_0^1 F(x) + S(1) =
(F(1)+S(1)) - \int_0^1 F(x) =
1 - \int_0^1 F(x) = 
\int_0^1 S(x)
$$
